src/engine/controls/cuecontrol.cpp

```diff
@@ -670,6 +802,69 @@ void CueControl::hotcueGotoAndPlay(HotcueControl* pControl, double v) {
     }
 }
 
+void CueControl::hotcueReloop(HotcueControl* pControl, double v) {
+    if (!v)
+        return;
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    CuePointer pCue(pControl->getCue());
+
+    // Need to unlock before emitting any signals to prevent deadlock.
+    lock.unlock();
+
+    if (!pCue) {
+        return;
+    }
+
+    double startPosition = pCue->getPosition();
+    if (startPosition == -1) {
+        return;
+    }
+
+    double endPosition = startPosition + pCue->getLength();
+    if (startPosition >= endPosition) {
+        return;
```

Comment:
This prevents us from removing a previous loop with an invalid. Is this intended here than please comment. If not this check can be removed.

Reply:
Are you referring to the case that loop is playing and you trigger hotcueReloop for a hotcue control that does not point to a valid loop so that nothing happens? Yes that is intended.

Reply:
I don't think that code exists anymore.